solver/llbsolver/bridge.go

```diff
@@ -124,88 +123,44 @@ func (b *llbBridge) loadResult(ctx context.Context, def *pb.Definition, cacheImp
 		if err := b.eachWorker(func(w worker.Worker) error {
 			return w.PruneCacheMounts(ctx, ids)
 		}); err != nil {
-			return nil, nil, err
-		}
-	}
-
-	res, bi, err := b.builder.Build(ctx, edge)
-	if err != nil {
-		return nil, nil, err
-	}
-	return res, bi, nil
-}
-
-func (b *llbBridge) Solve(ctx context.Context, req frontend.SolveRequest, sid string) (res *frontend.Result, err error) {
```

Comment:
I think we should still implement `Solve` for `llbBridge` so that it continues to implement `FrontendLLBBridge`, unless there's a specific reason it's been left out?

The way I read the intention of the code: `provenanceBridge` is a wrapper for `llbBridge` that collects all of the relevant provenance as well as performing the standard Solve.

Reply:
`llbBridge` does not have a `Solve` anymore and we shouldn't just add a function if there is nothing that calls it. Maybe `llbBridge` is not the best name for it anymore? The LLB builds still happen through `llbBridge` but not through the `Solve` function signature but the `loadResult` function that is called by the `ResultProxy.Result()` and calls into to solve graph with proper cache config.

Reply:
Theoretically `provenanceBridge` can be also seen like a `frontendBridge` if that helps.